File: vseros/2016-17/school/mountain.py
n = int(input())

start = 0
finish = 0
minstart = 0
minfinish = 0

prev = int(input())
for i in range(2, n + 1):
    top = int(input())
    if prev < top:
        start = i - 1
    if prev > top:
        finish = i
    if start * finish != 0 and finish > start and (minfinish - minstart == 0 or finish - start < minfinish - minstart)
        minstart = start
        minfinish = finish
    prev = top

if min == 0:
    print(0)
else:
    print(minstart, minfinish)

File: vseros/2016-17/school/mountain.cpp
#include <iostream>

using namespace std;

int main()
{
	int n;
	cin >> n;

	int up = 0;
	int prev = 0;
	int minfinish = 1;
	int minstart = 1;
	int start, finish;

	int top;
	cin >> top;

	for(int i = 2; i < n + 1; ++i)
	{
	    prev = top;
		cin >> top;

	    if (up == 1)
		{
	        if (top > prev)
	            start = i - 1;

	        if (top < prev)
			{
	            finish = i;
	            up = 0;
	            if (minfinish - minstart == 0 or finish - start < minfinish - minstart)
				{
	                minfinish = finish;
	                minstart = start;
				}
			}
		}

	    if (up == 0 && top > prev)
		{
            start = i - 1;
            up = 1;
		}
	}

	if (minfinish - minstart == 0)
	    cout << 0 << endl;
	else
	    cout << minstart << " " << minfinish << endl;
	
	return 10;
}
